### ISSUE TITLE
Video player doesn't load or play movies

### ISSUE THREAD

**Issue by Igglybuff** [dularion/streama#206]
Using beta v1.0.5 with the MySQL db on a fresh Debian 8 system.

TV shows work fine, but adding a local file to a newly created movie doesn't seem to register. The file appears in the File Manager and can be played from there but can't be played by clicking the play button on the dashboard item for that film.

Grails spits out this:

```
ERROR org.apache.catalina.core.ContainerBase.[Tomcat].[localhost] - Exception Processing ErrorPage[errorCode=404, location=/error] java.lang.IllegalStateException: There was a problem retrieving the current GrailsWebRequest. This usually indicates a filter ordering issue (the 'springSecurityFilterChain' filter-mapping element must be positioned after the 'grailsWebRequest' element when using @Secured annotations).
        at grails.plugin.springsecurity.web.access.intercept.AnnotationFilterInvocationDefinition.determineUrl(AnnotationFilterInvocationDefinition.groovy:90)
        at grails.plugin.springsecurity.web.access.intercept.AbstractFilterInvocationDefinition.getAttributes(AbstractFilterInvocationDefinition.groovy:72)
        at org.springframework.security.access.intercept.AbstractSecurityInterceptor.beforeInvocation(AbstractSecurityInterceptor.java:196)
        at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:123)
        at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:90)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330)
        at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:114)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330)
        at grails.plugin.springsecurity.web.filter.GrailsAnonymousAuthenticationFilter.doFilter(GrailsAnonymousAuthenticationFilter.groovy:53)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330)
        at org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:149)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330)
        at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:169)
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330)

```

Perused previous issues and the problem is very similar to [this](https://github.com/dularion/streama/issues/202). Googling yielded [this](https://github.com/grails-plugins/grails-spring-security-core/issues/412).

**Comment by gkiko** [comment]
I try to trigger the event that causes exception but can't reproduce it. Can you describe step by step how to do it.

Also, I found that uploading video suffers from a lot of bugs :bug:

**Comment by Igglybuff** [comment]
The video files already exist on the server, so I use the new Local File button to attach a video file to a newly created Movie entry. Clicking "Choose" next to the correct .mp4 behaves as expected, the green confirmation box appears then disappears as it should, but nothing appears on the Movie's page in the Manage Content > Movies page for that movie after clicking "Close" on the file selection dialog box. However, the chosen file does appear listed in the File Manager in the Manage Content section after doing this. Reloading the Movie's content management page brings back this message:

![novideofile](https://cloud.githubusercontent.com/assets/8899226/17508643/dd8bfe8a-5e0d-11e6-87a9-360932956046.png)

The Dashboard will show the movie as available, but since the video file failed to save/attach to the Movie entry the player doesn't load and I just get a black screen.

**Comment by dularion** [comment]
Im looking into that now. Will find out more soon.

**Comment by dularion** [comment]
fixed in  v1.0.6
